### DISCUSSION TITLE
테스트를 위해서, repository에 메서드를 추가하는 것에 대해

### DISCUSSION THREAD

**Discussion by hong-sile** [woowacourse-teams/2023-emmsale#512] [Ideas]
백엔드 분들 질문이 있슴다.

테스트에서만 사용하는 , JPARepository의 메서드를 정의해야하는 상황이 나오는데, 어떻게 해야할까요?
자세한 상황은 다음과 같습니다.

message를 추가하는 기능을 테스트하려고, 해서 when에서 message를 추가하고, then에서 message를 조회해와서 제대로 추가 됬는지 검증하려고합니다.

하지만 이 때 message에 대한 id가 없어서 별도의 메서드를 messageRepository에 정의를 해야합니다.
(단 별도의 쿼리를 작성하진 않습니다.)

이렇게 추가한 메서드가 실제로 프로덕션에서 호출될 일은 없는 상황인데, 이런 경우엔 어떻게 해야할까요?

**Comment by hong-sile** [comment]
```
--given 절 ---
//when
messageCommandService.sendMessage(request, member);
//then
//여기서 생성된 메시지를 검증해야 합니다.
```

then 절에서 생성된 메시지를 검증해야 하는데, 이를 검증하는 방법으론
message를 조회해서 내가 요청한 값으로 잘 저장됬는지 확인해야 합니다.

messageRepository.isExsistContent(requester.getContnet);
이런 새로운 메서드가 추가될것 같아요.

**Comment by amaran-th** [comment]
혹시 제가 이해한 대로라면 현재 요구사항(프로덕션 코드 상에서)에는 메시지 단건 조회(or 존재여부 확인)를 하는 상황이 없어서 문제가 되는 것이 맞나요??

**Reply by hong-sile** [reply]
넵 맞습니다.

**Reply by amaran-th** [reply]
생각을 해봤는데 아직 디자인이 나오지 않았지만, 에타를 보면 쪽지함에서 미리보기로 쪽지함별  **가장 최근 메시지의 내용**을 보여주잖아요. 저는 이게 기획적으로 필요할 것 같다고 생각하고 있었는데, 만약 이걸 요구사항으로 반영한다면 메시지에 대한 단건조회가 필요해질 수도 있으니 말씀하신 문제도 함께 해결되지 않을까 싶습니다!

**Reply by amaran-th** [reply]
비단 이 문제때문에 기획을 변경하자는 게 아니라, 저도 염두해두고 있던 기획이라 반영된다면 겸사겸사 해당 문제도 해결되지 않을까 싶어 말씀드려봅니당
+노션에 정리한 디자인 요구사항에도 포함되어 있습니다!

**Reply by hong-sile** [reply]
메시지에 대한 단건조회는 다른 방법으로 우르가 풀어내서 상관없습니다.

**Reply by hong-sile** [reply]
우르가 옆에서 findAll을 추천해주셨네요. findAll로 충분히 풀어낼 수 있을 것 같아요.
